File: ruby/PhoneNumberOrdering/create_subscriptions.rb
# create_subscriptions.rb
#
# A small script that sets up subscription callbacks for Bandwidth number ordering
#
# @copyright Bandwidth INC
require 'ruby-bandwidth-iris'

begin
    IRIS_ACCOUNT_ID = ENV.fetch("IRIS_ACCOUNT_ID")
    IRIS_USERNAME = ENV.fetch("IRIS_USERNAME")
    IRIS_PASSWORD = ENV.fetch("IRIS_PASSWORD")
    SITE_ID = ENV.fetch("SITE_ID")
rescue
    puts "Please set the environmental variables defined in the README"
    exit(-1)
end

if ARGV.length < 1
    puts "usage: ruby create_subscriptions.rb <base_url>"
    exit(-1)
end

base_url = ARGV[0]
if base_url[-1] != '/' then
    base_url += '/'
end

order_url = base_url + "subscriptions/orders" 
disconnects_url = base_url + "subscriptions/disconnects"

BandwidthIris::Client.global_options = {
    :account_id => IRIS_ACCOUNT_ID,
    :username => IRIS_USERNAME,
    :password => IRIS_PASSWORD,
    :api_endpoint => "https://dashboard.bandwidth.com/api"
}

subscription = {
  :order_type => "orders",
  :callback_subcription => {
    :URL => order_url 
  }
}
BandwidthIris::Subscription.create(subscription)

subscription = {
  :order_type => "disconnects",
  :callback_subcription => {
    :URL => disconnects_url 
  }
}
BandwidthIris::Subscription.create(subscription)

File: php/PhoneNumberOrdering/create_subscriptions.php
<?php

/**
 * create_subscriptions.php
 *
 * A small script that sets up subscription callbacks for Bandwidth number ordering
 *
 * @copyright Bandwidth INC
 */

require 'vendor/autoload.php';

$IRIS_ACCOUNT_ID = getenv("IRIS_ACCOUNT_ID");
$IRIS_USERNAME = getenv("IRIS_USERNAME");
$IRIS_PASSWORD = getenv("IRIS_PASSWORD");
$SITE_ID = getenv("SITE_ID");

if($IRIS_ACCOUNT_ID == '' || $IRIS_USERNAME == '' || $IRIS_PASSWORD == '' || $SITE_ID == '') {
    echo "Please set the environmental variables defined in the README\n";
    exit(-1);
}

$base_url = $argv[1];

if ($base_url == '') {
    echo "usage: php create_subscriptions.php <base_url>\n";
    exit(-1);
}

if (substr($base_url, -1) != "/") {
    $base_url = $base_url . "/";
}

$orders_url = $base_url . "subscriptions/orders";
$disconnects_url = $base_url . "subscriptions/disconnects";

$client = new \Iris\Client($IRIS_USERNAME, $IRIS_PASSWORD, ['url' => 'https://dashboard.bandwidth.com/api/']);
$account = new \Iris\Account($IRIS_ACCOUNT_ID, $client);

$subscription = $account->subscriptions()->create([
    "OrderType" => "orders",
    "CallbackSubscription" => [
        "URL" => $orders_url
    ]
]);

$subscription = $account->subscriptions()->create([
    "OrderType" => "disconnects",
    "CallbackSubscription" => [
        "URL" => $disconnects_url
    ]
]);
